lib/beaker/shared/error_handler.rb

```diff
@@ -9,7 +9,7 @@ def report_and_raise(logger, e, msg)
       logger.pretty_backtrace(bt).each_line do |line|
         logger.error(line)
       end
-      raise e
+      raise e unless e.is_a? Beaker::DSL::FailTest
```

Comment:
@samwoods1 what are the consequences of this line of change? At first look, the rest of the PR seems unrelated to the `report_and_raise` method.